File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkLocation.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.location.Location;

@Entity
public class WalkLocation {
    @PrimaryKey(autoGenerate = true)
    long locationId;

    long walkId;
    double latitude;
    double longitude;
    long timestamp;

    public double getAccuracy() {
        return accuracy;
    }

    public void setAccuracy(double accuracy) {
        this.accuracy = accuracy;
    }

    double accuracy;

    public void setWalkId(long walkId) {
        this.walkId = walkId;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return String.format("%s,%s", getLatitude(),getLongitude());
    }
}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkLocation.kt
package com.example.macarus0.walkiewalkie.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class WalkLocation {
    @PrimaryKey(autoGenerate = true)
    var locationId: Long = 0

    var walkId: Long = 0
    var latitude: Double = 0.toDouble()
    var longitude: Double = 0.toDouble()
    var timestamp: Long = 0

    var accuracy: Double = 0.toDouble()

    override fun toString(): String {
        return String.format("%s,%s", latitude, longitude)
    }
}
